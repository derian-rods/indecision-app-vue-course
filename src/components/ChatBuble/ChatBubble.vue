<template>
  <!-- Example Message -->
  <div v-if="isSender" class="flex justify-end">
    <div class="bg-blue-200 text-black p-2 rounded-lg max-w-xs">
      {{ message }}
    </div>
  </div>
  <!-- Example Received Message -->
  <div v-else class="flex">
    <div class="bg-gray-300 text-black p-2 rounded-lg max-w-xs">
      <span class="Capitalize">{{ message }}</span>
      <img
        v-if="imageUrl"
        :src="imageUrl"
        loading="eager"
        width="300"
        height="300"
        alt="Avatar"
        class="w-52 h-52 object-cover rounded-md"
      />
    </div>
  </div>
</template>

<script setup lang="ts">
interface ChatBubbleProps {
  message: string;
  isSender: boolean;
  imageUrl?: string;
}
defineProps<ChatBubbleProps>();
</script>
