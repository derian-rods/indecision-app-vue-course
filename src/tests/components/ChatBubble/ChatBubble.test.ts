import { describe, expect, it } from 'vitest';
import { mount } from '@vue/test-utils';
import ChatBubble from '@/components/ChatBubble/ChatBubble.vue';

describe('ChatBubble Component', () => {
  it('should render the chat Bubble with correct message', () => {
    const message = 'Hello, this is a test message!';
    const wrapper = mount(ChatBubble, {
      props: {
        message,
        isSender: true,
      },
    });

    expect(wrapper.text()).toContain(message);
  });

  it('should match snapshot if sender is false', () => {
    const wrapper = mount(ChatBubble, {
      props: {
        message: 'Snapshot test message',
        isSender: false,
        imageUrl: 'https://example.com/avatar.png',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match snapshot if sender is true', () => {
    const wrapper = mount(ChatBubble, {
      props: {
        message: 'Snapshot test message',
        isSender: true,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
