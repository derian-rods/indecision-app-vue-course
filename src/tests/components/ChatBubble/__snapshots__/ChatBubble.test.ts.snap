// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChatBubble Component > should match snapshot if sender is false 1`] = `
"<!-- Example Message -->
<!-- Example Received Message -->
<div class="flex">
  <div class="bg-gray-300 text-black p-2 rounded-lg max-w-xs"><span class="Capitalize">Snapshot test message</span><img src="https://example.com/avatar.png" loading="eager" width="300" height="300" alt="Avatar" class="w-52 h-52 object-cover rounded-md"></div>
</div>"
`;

exports[`ChatBubble Component > should match snapshot if sender is true 1`] = `
"<!-- Example Message -->
<div class="flex justify-end">
  <div class="bg-blue-200 text-black p-2 rounded-lg max-w-xs">Snapshot test message</div>
</div>"
`;
